src/components/chats/ChatList/ChatList.tsx

```diff
@@ -82,7 +83,12 @@ function ChatListContent({
     }
   )
 
-  const comments = getPostQuery.useQueries(currentData)
+  const { data: blockedIds } = getBlockedIdsInRootPostIdQuery.useQuery(postId)
+  const filteredIds = useMemo(() => {
+    return currentData.filter((id) => !blockedIds?.includes(id))
```

Comment:
Better use Set and `has` method for it, because `includes` isnt good for big arrays

Reply:
extracted to util and hook using set